packages/babel-types/src/definitions/es2015.js

```diff
@@ -537,7 +538,10 @@ defineType("TemplateElement", {
   builder: ["value", "tail"],
   fields: {
     value: {
-      // todo: flatten `raw` into main node
+      validate: assertShape({
+        raw: assertValueType("string"),
+        cooked: assertValueType("string"),
```

Comment:
`cooked` may also be `null` in a tagged template:

```js
tag`foo\x`
```

Reply:
@jridgewell made `cooked` optional